codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -161,29 +161,17 @@ public B gracefulShutdownTimeoutMillis(long gracefulShutdownTimeoutMillis) {
         }
 
         /**
-         * Determine if the encoder should queue frames to honor the value set by
-         * {@link #encoderMaxConcurrentStreams(int)}.
-         */
-        public B encoderEnforceMaxConcurrentStreams(boolean encoderEnforceMaxConcurrentStreams) {
-            encoderMaxConcurrentStreams = -1;
-            return thisB();
-        }
-
-        private boolean encoderEnforceMaxConcurrentStreams() {
-            return encoderMaxConcurrentStreams >= 0;
-        }
-
-        /**
          * How many initial streams are allowed to exists concurrently. Frames will be queued if they would result in
          * creating a stream which would cause the number of existing streams to exceed this number.
-         * @see #encoderEnforceMaxConcurrentStreams(boolean)
+         *
+         * Use {@code -1} if you not want to enforce max concurrent streams.
          */
         public B encoderMaxConcurrentStreams(int encoderMaxConcurrentStreams) {
             // This bounds are enforced here because the builder makes assumptions about its valid range to determine
             // if it should be used.
-            if (encoderMaxConcurrentStreams < 0) {
+            if (encoderMaxConcurrentStreams < -1) {
```

Comment:
this clearly did not turn out as intended :cry: The idea was to introduce a method to be able to "turn off buffering streams". This would hide the internal details of how it is managed (i.e. `encoderMaxConcurrentStreams` to -1).

Can we change `public B encoderEnforceMaxConcurrentStreams(boolean encoderEnforceMaxConcurrentStreams)` to `public B encoderMaxConcurrentStreamsDisableBuffering()` (or something similar) so we don't have to have the user know the details of passing in `-1`, `Integer.MAX_VALUE`, or generally have to know what is or is not valid to turn off the buffering?